### ISSUE TITLE
Arm64 build not available for v1.0

### ISSUE THREAD

**Issue by alexfacciorusso** [FreeCAD/FreeCAD#17181]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

The Apple M1/M2 builds are available, but the Windows Arm64 builds (for PCs like the new Surface Pro and Surface Laptop with ARM64 CPU) aren't available.

It would be great to have native support and performance for those devices.

### Full version info

```shell
ND-
```

### Subproject(s) affected?

None

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by alexfacciorusso** [comment]
I will be on holidays in the next few weeks but if anybody can easily-ish integrate the builds, I can definitely test them on my "holidays laptop"

**Comment by hyarion** [comment]
While you wait for a native build:
It looks like arm64 windows can run x86_64 windows builds of freecad by installing the OpenCL and OpenGL Compatibility Pack from Microsoft Store: https://forum.freecad.org/viewtopic.php?style=5&p=676066#p676066

**Comment by maxwxyz** [comment]
@oursland FYI

**Comment by oursland** [comment]
Unfortunately, that's not possible at this time as the underlying dependency packages are not available.

See: https://github.com/conda/conda/issues/11472

Hopefully this will change in the future.

**Comment by alexfacciorusso** [comment]
Yes it runs by default on the latest windows 11 update for arm64.

Although, the difference between "runs" and "runs natively" is quite huge (the CPUs on those machines are quite beefed up, especially the Snapdragon X Elite version, but when running in x64 emulation mode, the performance isn't incredible, but yes it works).

It seems something's moving now though, and being Conda a big project, I hope we'll see builds for it soonish.
A lot of software are finally getting arm64 builds in the last month or so, possibly helped by Microsoft; in the end, if there's native apps, people will buy the newer copilot+ laptops, so it's a Microsoft priority to get support as much as possible IMHO.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by adrianinsaval** [comment]
getting this done will probably require lots of patching in many projects, I don't think it's doable in time for 1.0 release.

@chennes is there any chance of compiling libpack for arm64?

**Comment by chennes** [comment]
It is probably doable, though I did hardcode x86 is a couple of places.

**Comment by ArminiusTux** [comment]
> ..but the Windows Arm64 builds (for PCs like the new Surface Pro and Surface Laptop with ARM64 CPU) aren't available.
> It would be great to have native support and performance for those devices too.
> 

Indeed official support is always preferable, but a 3rd party solution can be better than nothing.

The [FreeCAD MSYS2 pre-built WinARM64 package](https://packages.msys2.org/packages/mingw-w64-clang-aarch64-freecad?repo=clangarm64) is partially working - there seems to be continuining [issues with certain Python components](https://github.com/msys2/MINGW-packages/pull/19156).

**Comment by hmartinez82** [comment]
The MSYS2 project already has it compiled for Windows on ARM64: https://packages.msys2.org/base/mingw-w64-freecad
Most of the work for official builds will be compiling 3rd party libraries for ARM64 first.

**Comment by hmartinez82** [comment]
After the latest PySide fix for 6.8.0 (the packages are built and staged for upload soon). I got this:
![image](https://github.com/user-attachments/assets/f1b7068a-8320-436c-95a3-416de1120d0e)

**Comment by hyarion** [comment]
Nice!
Would be nice to add arm64 builds as weeklies, but maybe skip it for 1.0, as it needs proper testing

**Comment by Reqrefusion** [comment]
Is there any new news here? @oursland @adrianinsaval

**Comment by chennes** [comment]
The LibPack is available for WoA now, so we can use that if pixi still doesn't work.

**Comment by chennes** [comment]
It looks like Pixi is basically going to be a non-starter for the foreseeable future, there are over a dozen dependencies in our toolchain that as of this writing have no WoA pixi-build availability. Based on my experience working to get the LibPack build working I would not hold out hope for those all getting resolved in the near future.

So, although it's annoying and inconvenient, I think we have no choice but to spool up a LibPack-based branch of our release process. This will probably remain ARM-only, and get labelled "experimental" even for a non-weekly release (due to its use of bleeding edge library versions). I'll get started on that, and make sure it remains a distinct, removable, branch of the release process, so that when eventually we can switch back to just plain pixi builds it's a simple matter of deleting the LibPack-based builds.

I'm marking this as a v26.3.0 release blocker to make sure it stays on my radar and to ensure we do get the WoA release out this year.

**Comment by c360e5f1** [comment]
is there anyway I can help with this? I have two Windows on Arm physical machines and I have the ability to create more as virtual machines I can provide for development / testing via remote access.

**Comment by chennes** [comment]
It's basically ready to go, I'm running a test build of it here: https://github.com/chennes/FreeCAD/actions/runs/29945841158

**Comment by deepakrajsolanki** [comment]
### Request: Prioritize official FreeCAD Windows ARM64 builds

Hi FreeCAD maintainers and community,

I’m writing to strongly encourage work on **official Windows ARM64 (aarch64) releases**, and to ship them for the FreeCAD community as soon as feasible (ideally as part of the 1.1.x line or the next stable).

#### Why this matters now
- Windows on ARM devices (Snapdragon X Elite/Plus laptops, Surface Pro/Laptop with ARM, etc.) are increasingly common for engineering, students, and content creators.
- FreeCAD already ships **native ARM** for **macOS (Apple Silicon)** and **Linux (aarch64 AppImage)**. Windows is the missing platform.
- x86_64 under Prism/emulation works as a stopgap, but native builds would give better performance, lower power use, and a first-class experience—especially for 3D viewport, CAM, FEM, and longer modeling sessions.
- Community options (MSYS2 `mingw-w64-clang-aarch64-freecad`, self-builds) help early adopters but are not a substitute for signed, official installers on the Download page.

#### What would help users
1. **Official Windows ARM64 installer + portable (.7z)** on https://www.freecad.org/downloads.php and GitHub Releases (same cadence as x86_64).
2. Clear status on blockers (LibPack/conda-forge ARM64 deps, CI runners, code signing) and a rough target (e.g. weekly/dev builds first, then stable).
3. Mention of Windows ARM support (or “in progress”) on the Download / wiki pages so users know it’s on the roadmap.

#### Offer to help
I’m happy to:
- Test preview/weekly ARM64 builds on Windows 11 ARM hardware
- Report bugs with full About info and repro steps
- Share feedback from real mechanical-design / hobby CAD workflows

Native Windows ARM64 FreeCAD would make a big difference for a growing part of the community. Thank you for all the work on 1.1 and for considering this priority.

Best regards  
Deepakraj Solanki
“Snapdragon X laptop / Windows 11 ARM”

**Comment by Craige** [comment]
Looking forward to this. Can't wait to see how much better FreeCAD runs when not under emulation!

**Comment by chennes** [comment]
This will be closed by PR #31514

**Comment by hmartinez82** [comment]
Awesome, is there a portable build already? I would love to test it.

**Comment by maxwxyz** [comment]
@hmartinez82 it should be available next Wednesday when the weekly builds.

**Comment by chennes** [comment]
The test build for that PR is here: https://github.com/chennes/FreeCAD/releases/download/experimental-windows-libpack-release-2026.07.22/FreeCAD_experimental-windows-libpack-release-2026.07.22-Windows-arm64-experimental.7z

It is unsigned though. The first "real" build will be the next weekly, which will launch next Tuesday.

**Comment by Craige** [comment]
Just flagging that this does not appear in the weekly build this week. Does something need to be updated to publish the new build asset?

**Comment by chennes** [comment]
No, the asset signing failed, see https://github.com/FreeCAD/FreeCAD/pull/31759

**Comment by chennes** [comment]
OK, have a look at the WoA build here:
https://github.com/FreeCAD/FreeCAD/releases/tag/weekly-2026.08.12

**Comment by Craige** [comment]
Amazing! Thanks @chennes for your hard work on this. FreeCAD runs so much better when not run under emulation.

Startup times went from >30s under emulation to barely 1s with the native build, and that's to say nothing of the performance of the program as you actually try to work in it.
